src/table/utils/model-store.ts

```diff
@@ -0,0 +1,17 @@
+const keyStore = (initialState: { [key: string]: EngineAPI.IField } = {}) => {
+  const sharedState = initialState;
+  const store = {
+    get: (key: string) => {
+      return sharedState[key];
+    },
+    set: (key: string, value: EngineAPI.IField) => {
+      sharedState[key] = value;
+    },
+  };
+
+  return store;
+};
+
+const modelStore = keyStore({});
+
+export default modelStore;
```

Comment:
I can test that. Ideally it would I think, that way we create only a minimal set of objects.
Think however that every object has its own object store

Reply:
Yeah I would agree with that, I think it does. Just not sure.

Reply:
One thing that comes to mind is that it might "never" be clean-up. If a user adds a field and then remove it. That field will remain in the store.

Another thing is that I wonder what happens when a user is on a sheet with a sn-table, moves to another sheet and than back to the sheet with the table again. Is that store than re-created?

Reply:
I'll have to do some testing around this. I assume the store gets recreated every time. 
Only way around that would probably be if we can have a store somewhere else (re-use the nebula setup?). Also not sure how much effort we should put in to that. Let me get @Caele take on this

Reply:
My guess is that it's fine. The same approach is used in picasso.js to store result of measured text.

If you're uncertain about this, it's possible create a singleton per component instance in the supernova definition, using the useMemo hook.

Reply:
Okay so it seems that the store is shared for all tables on the sheet. The store even not re-created when moving away from the sheet. I guess it is created when the table code is loaded for the first time.